srcCppCommon/PerftestPrinter.cxx

```diff
@@ -0,0 +1,483 @@
+                        serializeTime + deserializeTime
+                );
+            }
+
+            break;
+    }
+}
+
+void PerftestPrinter::print_throughput(unsigned long long last_msgs,
+        unsigned long long mps,
+        double mps_ave,
+        unsigned long long bps,
+        double bps_ave,
+        unsigned long long missing_packets,
+        float missing_packets_percent,
+        std::string outputCpu)
+{
+    switch (_outputFormat)
+    {
+        case CSV :
+            printf("%14d,%14llu,%11llu,%14.0lf,%9.1lf,%10.1lf, %12llu, %16.2lf",
+                    _dataLength,
+                    last_msgs, mps, mps_ave,
+                    bps * 8.0 / 1000.0 / 1000.0, bps_ave * 8.0 / 1000.0 / 1000.0,
+                    missing_packets,
+                    missing_packets_percent);
+            if(_showCPU){
+                printf(",%6s", outputCpu.c_str());
+            }
+            printf("\n");
+            break;
+
+        case JSON :
+            if(_controlJsonIntervals)
+                    _controlJsonIntervals = false;
+            else
+                printf(",");
+            printf("\n\t\t\t\t{\n"
+                    "\t\t\t\t\t\"length\": %d,\n"
+                    "\t\t\t\t\t\"packets\": %llu,\n"
+                    "\t\t\t\t\t\"packets/s\": %llu,\n"
+                    "\t\t\t\t\t\"packets/s_ave\": %1.2lf,\n"
+                    "\t\t\t\t\t\"mbps\": %1.1lf,\n"
+                    "\t\t\t\t\t\"mbps_ave\": %1.1lf,\n"
+                    "\t\t\t\t\t\"lost\": %llu,\n"
+                    "\t\t\t\t\t\"lost_percent\": %1.2lf",
+                    _dataLength,
+                    last_msgs, mps, mps_ave,
+                    bps * 8.0 / 1000.0 / 1000.0, bps_ave * 8.0 / 1000.0 / 1000.0,
+                    missing_packets,
+                    missing_packets_percent
+                );
+            if(_showCPU){
+                printf(",\n\t\t\t\t\t\"cpu\": %s", outputCpu.c_str());
+            }
+            printf("\n\t\t\t\t}");
+            break;
+
+        case LEGACY :
+            printf("Packets: %8llu  Packets/s: %7llu  Packets/s(ave): %7.0lf  "
+                    "Mbps: %7.1lf  Mbps(ave): %7.1lf  Lost: %5llu (%1.2f%%)",
+                    last_msgs, mps, mps_ave,
+                    bps * 8.0 / 1000.0 / 1000.0, bps_ave * 8.0 / 1000.0 / 1000.0,
+                    missing_packets,
+                    missing_packets_percent
+                );
+            if(_showCPU){
+                printf(" CPU %s %%", outputCpu.c_str());
+            }
+            printf("\n");
+            break;
+    }
+}
+
+void PerftestPrinter::print_throughput_summary(int length,
+        unsigned long long interval_packets_received,
+        unsigned long long interval_time,
+        unsigned long long interval_bytes_received,
+        unsigned long long interval_missing_packets,
+        float missing_packets_percent,
+        std::string outputCpu)
+{
+    switch (_outputFormat)
```

Comment:
Apply comment about format. We need to be consistent across the whole document.

Reply:
ok